compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6049,7 +6049,8 @@ TR::Register *commonLoadEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, i
 
 TR::Register *commonLoadEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, int32_t size, TR::Register *targetReg, TR::CodeGenerator *cg)
    {
-   bool needSync = (node->getSymbolReference()->getSymbol()->isSyncVolatile() && cg->comp()->target().isSMP());
+   TR::Symbol *sym = node->getSymbolReference()->getSymbol();
+   bool needSync = cg->comp()->target().isSMP() && (sym->isVolatile() || sym->isAcqRel());
```

Comment:
If I understand correctly, `sym->isVolatile()` implies `sym->isAcqRel()`.  If that's true, why not simply test `sym->isAcqRel()` rather than `sym->isVolatile() || sym->isAcqRel()` here and throughout?

Reply:
Good point. I'll update the encoding to reflect that and fix the tests.

Reply:
how do you `setAcqRel()`, i.e. according to what? presumably according to usage in certain varhandle access methods?
if that were true, you couldn't get rid of isVolatile() testing yet, because it has implications on language levels where volatile existed but the others didn't. i also assumed the setting isn't applicable globally (i.e. the setting applies to the usage-spot, i am thinking of the possibility of transparent/acqrel accesses to a varhandle in the same program).

Reply:
In the consumer for this change (https://github.com/eclipse-openj9/openj9/pull/20475), `setAcqRel()` is called when the `Unsafe.get*Acquire()` and `Unsafe.put*Release()` intrinsics are recognized (these intrinsics are used in the implementation of the varhandle access methods). With how unsafe shadow symbols are used in OpenJ9, the setting is only applied to the usage site, not globally.